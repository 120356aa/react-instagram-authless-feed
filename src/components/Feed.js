import React, { Component } from 'react';
import Instagram from './../lib/Instagram'
import Media from './Media';

class Feed extends Component {
  constructor(props) {
    super(props);

    this.state = { media: [] };
  }

  componentDidMount() {
    Instagram.getFeed(this.props.username)
      .then(media => this.setState({ media: media }))
  }

  render() {
    return (
      <div className={this.props.className}>
        {this.state.media.map((media, index) => (
          <Media key={index} src={media.src} url={media.url} alt={media.alt} />
        ))}
      </div>
    );
  }
}

export default Feed;
