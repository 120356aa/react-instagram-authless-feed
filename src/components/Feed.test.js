import React from "react";
import { create, act } from "react-test-renderer";
import Feed from "./Feed";

describe("#render", async () => {
  it("on success", async () => {
    const getFeedFn = (username) =>
      Promise.resolve([
        {
          url: "https://placeholder.com/640",
          src: "https://via.placeholder.com/640",
          alt: "640x640px image from placeholder.com",
        },
      ]);

    let component;

    await act(async () => {
      component = create(
        <Feed className="Feed" username="jamespaulmoriarty" getFeedFn={getFeedFn} />
      );
    });

    expect(component.toJSON()).toMatchSnapshot();
  });

  it("on error", async () => {
    const getFeedFn = (username) => Promise.reject(new Error("fail"));

    let component;

    await act(async () => {
      component = create(
        <Feed className="Feed" username="jamespaulmoriarty" getFeedFn={getFeedFn} />
      );
    });

    expect(component.toJSON()).toMatchSnapshot();
  });
});
