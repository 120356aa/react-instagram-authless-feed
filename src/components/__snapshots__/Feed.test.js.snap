exports[`#render on error 1`] = `
<div
  className="Feed" />
`;

exports[`#render on success 1`] = `
<div
  className="Feed">
  <a
    href="https://placeholder.com/640"
    rel="noopener"
    target="_blank">
    <img
      alt="640x640px image from placeholder.com"
      src="https://via.placeholder.com/640" />
  </a>
</div>
`;
